services/blockstorage/sync/state_processing_blocks.go

```diff
@@ -0,0 +1,69 @@
+package sync
+
+import (
+	"context"
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+)
+
+type processingBlocksState struct {
+	blocks  *gossipmessages.BlockSyncResponseMessage
+	logger  log.BasicLogger
+	storage BlockSyncStorage
+	sf      *stateFactory
+}
+
+func (s *processingBlocksState) name() string {
+	return "processing-blocks-state"
+}
+
+func (s *processingBlocksState) processState(ctx context.Context) syncState {
+	if ctx.Err() == context.Canceled {
+		return nil
+	}
+
+	if s.blocks == nil {
+		s.logger.Error("fatal error in block sync, received no blocks to processing blocks state")
+		return s.sf.CreateIdleState()
+	}
+
+	firstBlockHeight := s.blocks.SignedChunkRange.FirstBlockHeight()
+	lastBlockHeight := s.blocks.SignedChunkRange.LastBlockHeight()
+
+	s.logger.Info("committing blocks from sync",
+		log.Int("block-count", len(s.blocks.BlockPairs)),
+		log.Stringable("sender", s.blocks.Sender),
+		log.Stringable("first-block-height", firstBlockHeight),
+		log.Stringable("last-block-height", lastBlockHeight))
+
+	for _, blockPair := range s.blocks.BlockPairs {
+		_, err := s.storage.ValidateBlockForCommit(&services.ValidateBlockForCommitInput{BlockPair: blockPair})
+
+		if err != nil {
+			s.logger.Error("failed to validate block received via sync", log.Error(err))
+			break
+		}
+
+		_, err = s.storage.CommitBlock(&services.CommitBlockInput{BlockPair: blockPair})
+
+		if err != nil {
+			s.logger.Error("failed to commit block received via sync", log.Error(err))
+			break
```

Comment:
*should* we break? think about the scenario: I'm at block 4, I just got a chunk with blocks 5 to 10, but just before I got to this current state, I managed to commit block 5 via the consensus route. Should I drop blocks 6-10?

Reply:
conversely, if *any* block fails to validate I think it's fair to treat the entire chunk as byzantine and break

Reply:
anyway this is an optimization so maybe open an issue and move on rather than delay merging this PR

Reply:
will open an issue, commitblock failure does not have to be fatal

Reply:
opened issue #320